compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1174,13 +1174,27 @@ OMR::Z::TreeEvaluator::vsqrtEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register*
 OMR::Z::TreeEvaluator::vminEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+   TR::DataType dt = node->getDataType().getVectorElementType();
+   TR_ASSERT_FATAL_WITH_NODE(node, cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_FACILITY_ENHANCEMENT_1) ||
+                                       !(dt == TR::Double ||
+                                          dt == TR::Float),
+                                    "VFMIN is only supported on z14 onwards.");
+   return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, (dt == TR::Double || dt == TR::Float) ? TR::InstOpCode::VFMIN : TR::InstOpCode::VMN);
    }
 
 TR::Register*
 OMR::Z::TreeEvaluator::vmaxEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+   TR::DataType dt = node->getDataType().getVectorElementType();
+   TR_ASSERT_FATAL_WITH_NODE(node, cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_FACILITY_ENHANCEMENT_1) ||
+                                       !(dt == TR::Double ||
+                                          dt == TR::Float),
+                                    "VFMAX is only supported on z14 onwards.");
+   return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, (dt == TR::Double || dt == TR::Float) ? TR::InstOpCode::VFMAX : TR::InstOpCode::VMX);
```

Comment:
Can `vminEvaluator` and `vmaxEvaluator` call a common function to save some code footprint and avoid duplication?  Their implementations are identical except for the ultimate choice of InstOpCode.

Reply:
Yes I have refactored the evaluators to actually do a safe check in the common function which are used to generate instructions in https://github.com/eclipse/omr/pull/6618/commits/92e99ed96c9294f949b12983899fc70ba6d6b5df

I also noticed some other vector evaluators on Z which are doing the same thing and can get benefit with some refactoring, so opened up OMR issue to come back to it later (https://github.com/eclipse/omr/issues/6654)